lib/node/runner/isaac_state_transit_test.go

```diff
@@ -0,0 +1,419 @@
+// We can test that a node broadcast propose ballot or B(`EXP`) in ISAACStateManager.
+package runner
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"boscoin.io/sebak/lib/block"
+	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/consensus"
+)
+
+// 1. All 3 Nodes.
+// 2. Proposer itself.
+// 3. When `ISAACStateManager` starts, the node proposes ballot to validators.
+func TestStateINITProposer(t *testing.T) {
+	nodeRunners := createTestNodeRunner(3)
+
+	nr := nodeRunners[0]
+
+	recv := make(chan struct{})
+	b := NewTestBroadcaster(recv)
+	nr.SetBroadcaster(b)
+	nr.SetProposerCalculator(SelfProposerCalculator{})
+
+	nr.Consensus().SetLatestConsensusedBlock(genesisBlock)
+
+	conf := consensus.NewISAACConfiguration()
+	conf.TimeoutINIT = time.Hour
+	conf.TimeoutSIGN = time.Hour
+	conf.TimeoutACCEPT = time.Hour
+	conf.TimeoutALLCONFIRM = time.Hour
+
+	nr.SetConf(conf)
+
+	nr.StartStateManager()
+
+	<-recv
+	require.Equal(t, 1, len(b.Messages()))
+	for _, message := range b.Messages() {
+		// This message must be proposed ballot
+		b, ok := message.(block.Ballot)
+		require.True(t, ok)
+		require.Equal(t, nr.localNode.Address(), b.Proposer())
+	}
+}
+
+// 1. All 3 Nodes.
+// 2. Not proposer itself.
+// 3. When `ISAACStateManager` starts, the node waits a ballot by proposer.
+// 4. But TimeoutINIT is an hour, so it doesn't broadcast anything.
+func TestStateINITNotProposer(t *testing.T) {
+	nodeRunners := createTestNodeRunner(3)
+
+	nr := nodeRunners[0]
+
+	b := NewTestBroadcaster(nil)
+	nr.SetBroadcaster(b)
+	nr.SetProposerCalculator(TheOtherProposerCalculator{})
+
+	nr.Consensus().SetLatestConsensusedBlock(genesisBlock)
+
+	conf := consensus.NewISAACConfiguration()
+	conf.TimeoutINIT = time.Hour
+	conf.TimeoutSIGN = time.Hour
+	conf.TimeoutACCEPT = time.Hour
+	conf.TimeoutALLCONFIRM = time.Hour
+
+	nr.SetConf(conf)
+
+	nr.StartStateManager()
+	time.Sleep(1 * time.Second)
```

Comment:
Would be nice to have a `defer` statement that stops the StateManager, otherwise we'll end up with dangling StateManager all over the place in the tests.